components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -94,9 +104,19 @@ public void index(Patient patient)
 
         for (Feature phenotype : patient.getFeatures()) {
             input.addField((phenotype.isPresent() ? "" : "negative_") + phenotype.getType(), phenotype.getId());
+            VocabularyTerm properTerm = this.ontologyService.getTerm(phenotype.getId());
+            if (properTerm != null) {
+                Set<VocabularyTerm> parents = properTerm.getAncestorsAndSelf();
+                for (VocabularyTerm term : parents) {
+                    allAncestors.add(term.getId());
+                }
+            } else {
```

Comment:
Empty else branch? I'd suggest either logging something on debug or remove the branch.